### DISCUSSION TITLE
Do you have an example on how to Update/Migrate the IndexedDB Database

### DISCUSSION THREAD

**Discussion by rebelweb** [amuste/DnetIndexedDb#12] [Q&A]
I need to add a column to a model store. Do you have an example of how to update the database? I tried changing the version number before deploying and that didn't seem to help. I have checked the code out and tried to follow the logic and not finding how a database update is triggered.

**Comment by amuste** [comment]
By changing the version number, the indexedDb API fires the update event when the Open method is executed. With the current version of the library it works, no changes are necessary. To test it, follow the steps below. I have uploaded a branch where I have added a new method that will help you test the update. I have also added a new index to the DB model with name 'GridColumnData'

In this line I subscribe to the upgradeneeded event of the api:
https://github.com/amuste/DnetIndexedDb/blob/669371fe68cdbda68431b1271ed929d10ec4bb31/src/DnetIndexedDb/wwwroot/dnet-indexeddb.js#L105

The code that runs:
https://github.com/amuste/DnetIndexedDb/blob/669371fe68cdbda68431b1271ed929d10ec4bb31/src/DnetIndexedDb/wwwroot/dnet-indexeddb.js#L92

The branch is as follows:
Feature_20210104_Fix_Open_and_Upgrade

Steps
1- In the index.razor file, comment lines 81 and 83
2- Run the application.
3- Stop the application
4- In the Index.razor file in OnAfterRenderAsync comment line 29 and uncomment line 33
5- In the Startup.cs file, uncomment the TestParam index in GetGridColumnDatabaseModel and change the version number.
5- Run the application.
6. Verify in the Application tab of the browser that the new TestParam parameter has been created.
